File: tests/jnet/TestBytes.java
import java.util.Date;
import java.util.ArrayList;
import org.junit.Test;
import org.junit.Assert;
import org.junit.Assume;
import jnet.Bytes;
import jnet.MissingDataException;


public class TestBytes {

    @Test
    public void testIntToBytesAndBytesToInt() {
        for (int i = 0; i < TestMain.RANDOM_TEST_AMOUNT; i++) {
            int expected = TestMain.RANDOM_DIS.nextInt();
            byte[] bytes = Bytes.intToBytes(expected);
            int generated = Bytes.bytesToInt(bytes);

            Assert.assertEquals(generated, expected);
        }
    }

    @Test
    public void testBytesToIntWithNullArgument() {
        Assert.assertThrows(NullPointerException.class, () -> Bytes.bytesToInt(null));
    }

    @Test
    public void testBytesToIntWithLongArgument() {
        Assert.assertThrows(MissingDataException.class,
                            () -> Bytes.bytesToInt(new byte[] {0,1,2,3,4}));
    }

    @Test
    public void testBytesToIntWithShortArgument() {
        Assert.assertThrows(MissingDataException.class,
                            () -> Bytes.bytesToInt(new byte[] {0,1,2}));
    }

    @Test
    public void testSerDesString() {
        for (int i = 0; i < TestMain.RANDOM_TEST_AMOUNT; i++) {
            String expected = TestMain.randomString(100);
            byte[] bytes = Bytes.serialize(expected);
            String generated = (String) Bytes.deserialize(bytes);

            Assert.assertEquals(generated, expected);
        }
    }

    @Test
    public void testSerializeWithNullArgument() {
        Assert.assertThrows(NullPointerException.class, () -> Bytes.serialize(null));
    }

    @Test
    public void testDeserializeWithNullArgument() {
        Assert.assertThrows(NullPointerException.class, () -> Bytes.deserialize(null));
    }

    @Test
    public void testSerDesJavaUtils() {
        Date dateExpected = new Date();
        byte[] dateBytes = Bytes.serialize(dateExpected);
        Date dateGenerated = (Date) Bytes.deserialize(dateBytes);
        Assert.assertEquals(dateGenerated, dateExpected);

        ArrayList<String> listExpected = new ArrayList<>();
        for (int i = 0; i < TestMain.RANDOM_TEST_AMOUNT; i++) {
            listExpected.add(TestMain.randomString(10));
        }
        byte[] listBytes = Bytes.serialize(listExpected);
        ArrayList<String> listGenerated = (ArrayList<String>) Bytes.deserialize(listBytes);
        Assert.assertEquals(listGenerated, listExpected);
    }

    @Test
    public void testStringToBytesAndBytesToString() {
        for (int i = 0; i < TestMain.RANDOM_TEST_AMOUNT; i++) {
            String expected = TestMain.randomString(100);
            byte[] bytes = Bytes.stringToBytes(expected);
            String generated = Bytes.bytesToString(bytes);

            Assert.assertEquals(generated, expected);
        }
    }

    @Test
    public void testStringToBytesWithEmptyArgument() {
        Assert.assertArrayEquals(Bytes.stringToBytes(""), new byte[] {});
    }

    @Test
    public void testBytesToStringWithEmptyArgument() {
        Assert.assertEquals(Bytes.bytesToString(new byte[] {}), "");
    }

    @Test
    public void testStringToBytesWithNullArgument() {
        Assert.assertThrows(NullPointerException.class, () -> Bytes.stringToBytes(null));
    }

    @Test
    public void testBytesToStringWithNullArgument() {
        Assert.assertThrows(NullPointerException.class, () -> Bytes.bytesToString(null));
    }

}

File: tests/pnet/test_bytes.py
import pytest
import conftest
from pnet import byteutils
from pnet.error import MissingDataError


class TestBytes:

    def test_int_to_bytes_and_bytes_to_int(self):
        for i in range(conftest.RANDOM_TEST_AMOUNT):
            expected: int = conftest.random_int32()
            b: bytes = byteutils.int_to_bytes(expected)
            generated: int = byteutils.bytes_to_int(b)

            assert generated == expected

    def test_bytes_to_int_with_null_argument(self):
        with pytest.raises(TypeError):
            byteutils.bytes_to_int(None)

    def test_bytes_to_int_with_long_argument(self):
        with pytest.raises(MissingDataError):
            byteutils.bytes_to_int(bytes([0, 1, 2, 3, 4]))

    def test_bytes_to_int_with_short_argument(self):
        with pytest.raises(MissingDataError):
            byteutils.bytes_to_int(bytes([0, 1, 2]))

    def test_ser_des_string(self):
        for i in range(conftest.RANDOM_TEST_AMOUNT):
            expected: str = conftest.random_str(100)
            b: bytes = byteutils.str_to_bytes(expected)
            generated: str = byteutils.bytes_to_str(b)

            assert generated == expected

    def test_str_to_bytes_with_empty_argument(self):
        assert byteutils.str_to_bytes("") == bytes()


    def test_bytes_to_str_with_empty_argument(self):
        assert byteutils.bytes_to_str(bytes()) == ""

    def test_str_to_bytes_with_null_argument(self):
        with pytest.raises(TypeError):
            byteutils.str_to_bytes(None)

    def test_bytes_to_str_with_null_argument(self):
        with pytest.raises(TypeError):
            byteutils.bytes_to_str(None)
